src/control/server/iosrv_test.go

```diff
@@ -0,0 +1,64 @@
+package server
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+)
+
+func TestFindBinaryInPath(t *testing.T) {
```

Comment:
Just a philosophical quibble, but I'm not wild about having to manipulate real filesystems and environment variables in unit tests. I'd prefer to move toward using interfaces and dependency injection in the Go code to make it easy to mock these kinds of things.

Reply:
This kind of testing is how the stdlib tests itself... How would we test this using interfaces?

Reply:
The strategy I like is creating a thin interface layer over the external packages. The default implementation just calls the real function, but a mock version can be injected specifically for the tests. In this case I could see calls like os.Readlink and os.Stat being mocked. Allows for easy testing of various failure paths.